### PR TITLE
refactor(shared-docs) improve shiki integration

### PR DESCRIPTION
**Pull request by JeanMeche**
Replaces the shiki css variable implementation with the inline dual-theme
Also uses shiki for API docs pages

### CODE REVIEW

File: bazel/api-gen/test/BUILD.bazel [hunk 1]
```diff
@@ -1,22 +1,20 @@
-load("//bazel/api-gen:generate_api_docs.bzl", "generate_api_docs")
+# generate_api_docs(
```
Comment (JeanMeche) [inline bazel/api-gen/test/BUILD.bazel:1]: I might need help on this one. 

Those tests fail with 

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /private/var/tmp/_bazel_matthieu.riegler/18ba3444c4a04ad1be50f3a474964109/sandbox/darwin-sandbox/805/execroot/dev-infra/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/bazel/api-gen/rendering/render_api_to_html.sh.runfiles/npm/node_modules/shiki/dist/index.mjs not supported.
Instead change the require of /private/var/tmp/_bazel_matthieu.riegler/18ba3444c4a04ad1be50f3a474964109/sandbox/darwin-sandbox/805/execroot/dev-infra/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/bazel/api-gen/rendering/render_api_to_html.sh.runfiles/npm/node_modules/shiki/dist/index.mjs to a dynamic import() which is available in all CommonJS modules.
```

File: bazel/api-gen/generate_api_docs.bzl [hunk 2]
```diff
@@ -15,7 +15,7 @@ def generate_api_docs(name, module_name, entry_point, srcs, import_map = {}, ext
         extra_entries = extra_entries,
     )
 
-    render_api_to_html(
```
Comment (JeanMeche) [inline bazel/api-gen/generate_api_docs.bzl:18]: I might need help on this one. 

Those tests fail with 

`Error [ERR_REQUIRE_ESM]: require() of ES Module /private/var/tmp/......./bin/bazel/api-gen/rendering/render_api_to_html.sh.runfiles/npm/node_modules/shiki/dist/index.mjs not supported.
Instead change the require of /private/var/tmp/...../bin/bazel/api-gen/rendering/render_api_to_html.sh.runfiles/npm/node_modules/shiki/dist/index.mjs to a dynamic import() which is available in all CommonJS modules.`

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit d30a56c19bafaac67cf44e605ed8c2c0e45b0a51.

The changes were merged into the following branches: main

**Pull request closed without merging**